packages/x-date-pickers-pro/src/DateRangePickerDay/DateRangePickerDay.tsx

```diff
@@ -226,7 +226,7 @@ const DateRangePickerDayDay = styled(PickersDay, {
   ...(!ownerState.selected && {
     backgroundColor: 'transparent',
   }),
-  ...(!ownerState.isHighlighting && {
+  ...(!ownerState.isHighlighting && !ownerState.isDragging && {
```

Comment:
@joserodolfofreitas This should have fixed the issue you noticed with grey border being applied to the drag start day when you drag away from it.